metricbeat/module/kafka/consumergroup/query.go

```diff
@@ -0,0 +1,210 @@
+package consumergroup
+
+import (
+	"sync"
+
+	"github.com/Shopify/sarama"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/metricbeat/module/kafka"
+)
+
+type client interface {
+	ListGroups() ([]string, error)
+	DescribeGroups(group []string) (map[string]kafka.GroupDescription, error)
+	FetchGroupOffsets(group string) (*sarama.OffsetFetchResponse, error)
+}
+
+func fetchGroupInfo(
+	emit func(common.MapStr),
+	b client,
+	groupsFilter, topicsFilter func(string) bool,
+) error {
+	type result struct {
+		err   error
+		group string
+		off   *sarama.OffsetFetchResponse
+	}
+
+	groups, err := listGroups(b, groupsFilter)
+	if err != nil {
+		logp.Err("failed to list known kafka groups: %v", err)
+		return err
+	}
+	if len(groups) == 0 {
+		return nil
+	}
+
+	debugf("known consumer groups: ", groups)
+
+	wg := sync.WaitGroup{}
+	results := make(chan result, len(groups))
+	for _, group := range groups {
+		group := group
```

Comment:
Yes, as `group` is used inside the go-routine. Copy is used to capture the `group` in the local scope.

Alternatively one can write:

```
for group := range groups {
  go func(group string) {
   ...
  }(group)
}
```

which is similar to writing:

```
for group := range groups {
  group := group
  go func() {
   ...
  }()
}
```